aead/src/lib.rs

```diff
@@ -68,96 +75,269 @@ impl fmt::Display for Error {
 
 impl core::error::Error for Error {}
 
-/// Nonce: single-use value for ensuring ciphertexts are unique
-pub type Nonce<A> = Array<u8, <A as AeadCore>::NonceSize>;
-
-/// Tag: authentication code which ensures ciphertexts are authentic
-pub type Tag<A> = Array<u8, <A as AeadCore>::TagSize>;
-
-/// Authenticated Encryption with Associated Data (AEAD) algorithm core trait.
-///
-/// Defines nonce, tag, and overhead sizes that are consumed by various other
-/// `Aead*` traits.
-pub trait AeadCore {
+/// Authenticated Encryption with Associated Data (AEAD) algorithm trait.
+pub trait Aead {
     /// The length of a nonce.
     type NonceSize: ArraySize;
 
-    /// The maximum length of the tag.
+    /// The length of a tag.
     type TagSize: ArraySize;
 
-    /// The upper bound amount of additional space required to support a
-    /// ciphertext vs. a plaintext.
-    type CiphertextOverhead: ArraySize + Unsigned;
+    /// Constant which defines whether AEAD specification appends or prepends tags.
+    ///
+    /// It influences the behavior of the [`Aead::encrypt_to_vec`], [`Aead::decrypt_to_vec`],
+    /// [`Aead::encrypt_to_buffer`], and [`Aead::decrypt_to_buffer`] methods.
+    ///
+    /// If the specification does not explicitly specify tag kind, we default to postfix tags.
+    const IS_POSTFIX: bool = true;
```

Comment:
I definitely prefer the approach in #1714 of using a marker trait for postfix-tagged AEADs and using that to gate a blanket impl, especially since we can't provide a good efficient implementation of prefix-tagged AEADs.

Reply:
A separate marker trait would result in a more complex API surface and artificial (in my opinion) restriction on using the postfix methods. Right now we have everything in one trait with easy-to-read docs.

Reply:
This PR adds hundreds of lines of code and dozens of new methods without adding any real new functionality, other than `inout` integration which should only take a few dozen lines of code.

Compared to #1714, the API surface is vastly more complex.

Reply:
Most of the added methods are ergonomic `inplace` and `to_buf` methods. 

>few dozen lines of code

Can you show these "few dozen lines"? Because unless you want to remove the helper methods, I don't see how exactly you plan to do this.

Reply:
I have reduced the amount of code duplication in this [commit](https://github.com/RustCrypto/traits/pull/1713/commits/1e55bec6044b5c32a45143024a25b24f2064d58b). Does this address your concerns?